state/sqlserver/migration.go

```diff
@@ -108,23 +109,23 @@ func (m *migration) executeMigrations() (migrationResult, error) {
 		defer db.Close()
 	}
 
-	err = m.ensureSchemaExists(db)
+	err = m.ensureSchemaExists(ctx, db)
```

Comment:
Is there a way all these `ensure` statements can be put in a lock? Otherwise there could be race conditions when multiple sidecars are starting up at the same time.

Reply:
Uhm looks like most (all) statements perform atomic queries with `IF NOT EXIST`, so it's probably ok.